src/main/scala/com/advancedtelematic/campaigner/http/UpdateResource.scala

```diff
@@ -48,8 +52,18 @@ class UpdateResource(extractNamespace: Directive1[Namespace])
     HypermediaResource(links, update)
   }
 
+  val groupUpdateResolver = new GroupUpdateResolver(deviceRegistry, resolver)
+
+  import com.advancedtelematic.libats.http.UUIDKeyAkka._
+
   val route: Route =
     extractNamespace { ns =>
+      pathPrefix("group_updates" / GroupId.Path) { groupId =>
```

Comment:
@jerrytrieu what about `group_updates/<group_id>` ?

Reply:
Idiomatic way in this case would be  `/updates?groupId=<group_id>`

Reply:
That is what I hard at first...

Reply:
That looks too simple for a complex query. Should be explicit `updates_for` or `applicable_updates`.

Reply:
@jerrytrieu `/updates?groupId=<group_id>` says exactly this.
IMO the best way would be `/campaigns?groupId=<group_id>` and returning an update as part of a campaign aggregate. The reason is there is no relation between an update and an update outside of context of a campaign

Reply:
*had

I don't think the api url should say how hard it is to get the resources, or any other detail of the implementation for that matter. 

@koshelev That would complicate getting the results but I guess would be possible... you mean something like `/campaigns/<campaign_id>?groupId=<group_id>` ? Since we already had get on `updates` I thought just returning that result would be fine.

Reply:
@simao If I understand it correctly, the endpoint should return all updates ever applied to a group. As every campaign is the link between an update and a group, we don't need a `campaign_id`. Or am I missing something?

Reply:
ah I see what you mean, yes ok..

Reply:
Ok, so I misunderstood what the endpoint is for. It should return **possible** updates for a group, not already scheduled updates. So `/updates?groupId=<group_id>` should do.

Reply:
So... should I revert to what I had at first or?

Reply:
AFAIU you had `/updates?groupId=<group_id>`, right? In this case it would be great if you revert.

Reply:
@jerrytrieu  ?

Reply:
I don't see `GET /updates?groupId=id` and think `returns all updates that apply to all vehicles in a device group by querying an external resolver`. The information for that query doesn't exist under `/updates` resource. But I agree about not thinking about implementation in the API uri, so okay.

Reply:
hm I see your point.. we could add `applicable_updates` then we do `applicable_updates?groupId=<group_id>` this would expose the  applicable_updates resource which kind of makes sense.

Reply:
We can. But it feels like documenting endpoints with URIs. And as I mentioned, it is unlikely we will need to request applied updates outside of campaign's context.
What about a compromise: `GET /updates?groupId=<id>&applicable` and `GET /updates?groupId=<id>&applied`?